File: frontend/src/types/purchaseOrder.ts
export interface PurchaseOrderProduct {
  id?: number;
  productId?: number;
  productName?: string;
  product?: {
    id: number;
    name: string;
  };
  quantity: number;
  expectedPallets: string;
  price: number;
}

export interface Pallet {
  id?: number;
  palletName: string;
  quantity: number;
  maximumCapacity: number;
  status: string;
  manufacturingDate: string;
  expiryDate: string;
  supplierName: string;
  productId?: number;
  productName?: string;
  product?: {
    id: number;
    name: string;
  };
  purchaseOrderId?: number;
}

export interface PurchaseOrder {
  id?: number;
  supplierName: string;
  userId?: string;
  expectedArrivalTime: string;
  totalPrice: number;
  status: "Pending" | "Processing" | "Ready to Ship" | "Shipping" | "QQ CHECK" | "Complete";
  products: PurchaseOrderProduct[];
  pallets: Pallet[];
  createdAt?: string;
  createdBy?: string;
  notes?: string;
}

export type PurchaseOrderStatus = "Pending" | "Processing" | "Ready to Ship" | "Shipping" | "QQ CHECK" | "Complete" | "Rejected";

// Enhanced interfaces for Supply Manager features
export interface SupplierInfo {
  id: string;
  name: string;
  contactEmail?: string;
  contactPhone?: string;
  leadTimeDays?: number;
  minimumOrderValue?: number;
  rating?: number;
  isActive: boolean;
}

File: backend/src/main/java/com/project/warehouse_management_system/PurchaseOrder/model/PurchaseOrder.java
package com.project.warehouse_management_system.PurchaseOrder.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.project.warehouse_management_system.Inventory.model.Pallet;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Setter
@Getter
@Entity
@EntityListeners(AuditingEntityListener.class)
@Table(name = "purchase_orders")
public class PurchaseOrder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String supplierName;

    @Column(name = "user_id")
    private String userId;

    @Column(nullable = false)
    private Date expectedArrivalTime;

    @Column(nullable = false)
    private Double totalPrice;

    @Column(nullable = false)
    private String status = "Pending"; // Pending, Processing, Ready to Ship, Shipping, QQ CHECK, Complete, Rejected

    @OneToMany(targetEntity = PurchaseOrderProduct.class, cascade = CascadeType.ALL)
    @JoinColumn(name = "purchase_order_id", referencedColumnName = "id")
    private List<PurchaseOrderProduct> products = new ArrayList<>();

    @OneToMany(mappedBy = "purchaseOrder", cascade = CascadeType.ALL)
    private List<Pallet> pallets = new ArrayList<>();

    @Column(length = 1000)
    private String notes;

    @CreatedDate
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @LastModifiedDate
    @JsonIgnore
    @Column(name = "last_modified", insertable = false)
    private LocalDateTime lastModified;
}